### DISCUSSION TITLE
Method to lex a single token (without context)

### DISCUSSION THREAD

**Discussion by asvarga** [tree-sitter/tree-sitter#2377] [Q&A]
Does tree-sitter expose a method to lex a single token from an input? I understand that TS lexing is "context-aware", so I'm looking for some way to do this without context. The use case is that I'm trying to do some pre-processing of tokens before parsing. Thanks!

**Comment by ahelwer** [comment]
Are you using tree-sitter as a lexing step before your own parser? Not really supported.

**Reply by asvarga** [reply]
I was planning on using tree-sitter for parsing as well. I'd like to be able to add annotations to an existing language and find and replace those annotations in code. However I want to ignore annotations when they're inside a string or are otherwise legal in the original grammar. The processing of these annotations is complex so this can't be accomplished by just modifying the original grammar.

**Reply by ahelwer** [reply]
Have you looked into whether this can be done by adding logic to the external scanner?

**Reply by asvarga** [reply]
I've ended up just modifying a separate external lexer for processing prior to parsing for a proof of concept, but may try properly integrating it with T-S in the future.

**Comment by asvarga** [comment]
This thread seems to answer my question in the negative: https://github.com/tree-sitter/tree-sitter/discussions/1036
